From 4017ba2cb2c3b4dd574eb501190f3f0eceb53115 Mon Sep 17 00:00:00 2001
From: Wajahat Khan <w-khan@ti.com>
Date: Tue, 28 Jun 2011 23:32:30 -0500
Subject: [PATCH] OMAP4: HDMI: Correct behavior on Syslink IPC_STOP event

When the Syslink IPC_STOP is received, the correct behavior should be
_WAIT_FOR_IPC because we are waiting for the Syslink notify to be ready.

Also, give a more descriptive name to the state in which the HDMI
driver is not registered with the IPC notifier.

Signed-off-by: Wajahat Khan <w-khan@ti.com>
Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
Signed-off-by: Juan Gutierrez <jgutierrez@ti.com>

---
 arch/arm/plat-omap/hdmi_lib.c              |    9 +++++----
 arch/arm/plat-omap/include/plat/hdmi_lib.h |    2 +-
 2 files changed, 6 insertions(+), 5 deletions(-)

Index: kernel/arch/arm/plat-omap/hdmi_lib.c
===================================================================
--- kernel.orig/arch/arm/plat-omap/hdmi_lib.c	2011-07-01 18:11:37.099653746 +0530
+++ kernel/arch/arm/plat-omap/hdmi_lib.c	2011-07-01 18:11:44.449649133 +0530
@@ -1522,7 +1522,7 @@
 {
 	int status = 0;
 	u16 *proc_id = (u16 *)v;
-	static u32 wa_state_last = HDMI_WA_STATE_NOTIFY_EVENT_NOTREG;
+	static u32 wa_state_last = HDMI_WA_STATE_NOTIFY_WAIT_FOR_IPC;
 
 	switch ((int)val) {
 	case IPC_START:
@@ -1553,7 +1553,7 @@
 			if (status == NOTIFY_S_SUCCESS) {
 				mutex_lock(&hdmi.wa_state_lock);
 				hdmi.wa_state =
-					HDMI_WA_STATE_NOTIFY_EVENT_NOTREG;
+					HDMI_WA_STATE_NOTIFY_WAIT_FOR_IPC;
 				mutex_unlock(&hdmi.wa_state_lock);
 			}
 		}
@@ -1795,7 +1795,8 @@
 #ifdef CONFIG_OMAP_HDMI_AUDIO_WA
 	if (omap_chip_is(hdmi.audio_wa_chip_ids)) {
 		mutex_lock(&hdmi.wa_state_lock);
-		if (hdmi.wa_state == HDMI_WA_STATE_NOTIFY_EVENT_NOTREG) {
+		if (hdmi.wa_state == HDMI_WA_STATE_IPC_NOTIFIER_NOTREG) {
+			pr_err("******IPC_REGISTER_NOTIFIER_CALL******\n");
 			r = ipc_register_notifier(&hdmi_syslink_notify_block);
 			hdmi.wa_state = HDMI_WA_STATE_NOTIFY_WAIT_FOR_IPC;
 		}
@@ -1838,7 +1839,7 @@
 	mutex_init(&hdmi.wa_state_lock);
 	mutex_init(&hdmi.wa_notify_lock);
 	mutex_lock(&hdmi.wa_state_lock);
-	hdmi.wa_state = HDMI_WA_STATE_NOTIFY_EVENT_NOTREG;
+	hdmi.wa_state = HDMI_WA_STATE_IPC_NOTIFIER_NOTREG;
 	mutex_unlock(&hdmi.wa_state_lock);
 	hdmi.audio_wa_chip_ids.oc = CHIP_IS_OMAP4430ES2 |
 			CHIP_IS_OMAP4430ES2_1 | CHIP_IS_OMAP4430ES2_2;
Index: kernel/arch/arm/plat-omap/include/plat/hdmi_lib.h
===================================================================
--- kernel.orig/arch/arm/plat-omap/include/plat/hdmi_lib.h	2011-07-01 18:11:17.212149117 +0530
+++ kernel/arch/arm/plat-omap/include/plat/hdmi_lib.h	2011-07-01 18:11:44.449649133 +0530
@@ -474,7 +474,7 @@
 #ifdef CONFIG_OMAP_HDMI_AUDIO_WA
 /* HDMI audio workaround states */
 enum hdmi_audio_wa_state {
-	HDMI_WA_STATE_NOTIFY_EVENT_NOTREG,
+	HDMI_WA_STATE_IPC_NOTIFIER_NOTREG,
 	HDMI_WA_STATE_NOTIFY_WAIT_FOR_IPC,
 	HDMI_WA_STATE_NOTIFY_EVENT_REG,
 	HDMI_WA_STATE_STOPPED,
